src/Host/Broker/Impl/RemoteUri/RemoteUriHelper.cs

```diff
@@ -5,37 +5,88 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Net;
+using System.Net.WebSockets;
+using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
 using Microsoft.Common.Core;
 using Microsoft.R.Host.Protocol;
 
 namespace Microsoft.R.Host.Broker.RemoteUri {
     public class RemoteUriHelper {
+
+        private static async Task DoWebSocketReceiveSendAsync(WebSocket receiver, WebSocket sender, CancellationToken ct) {
+            if (receiver == null || receiver == null) {
+                return;
+            }
+
+            ArraySegment<byte> receiveBuffer = WebSocket.CreateServerBuffer(65335);
+            while (receiver.State == WebSocketState.Open && sender.State == WebSocketState.Open) {
+                if (ct.IsCancellationRequested) {
+                    Task.WhenAll(
+                        receiver.CloseAsync(WebSocketCloseStatus.NormalClosure, "Normal closure", CancellationToken.None),
```

Comment:
There's no point in using `Task.WhenAll` here, since you're not actually waiting on it. On the other hand, if `CloseAsync` might throw, you need to `SilenceException` those.